presto-parser/src/main/java/io/prestosql/sql/parser/SqlParser.java

```diff
@@ -216,12 +217,14 @@ public void exitNonReserved(SqlBaseParser.NonReservedContext context)
             context.getParent().removeLastChild();
 
             Token token = (Token) context.getChild(0).getPayload();
-            context.getParent().addChild(new CommonToken(
+            Token newToken = new CommonToken(
                     new Pair<>(token.getTokenSource(), token.getInputStream()),
                     SqlBaseLexer.IDENTIFIER,
                     token.getChannel(),
                     token.getStartIndex(),
-                    token.getStopIndex()));
+                    token.getStopIndex());
+
+            context.getParent().addChild(new TerminalNodeImpl(newToken));
```

Comment:
I think we may want to call `Parser.createTerminalNode` instead of creating `TerminalNodeImpl` directly.

Reply:
Looks like `Parser.createTerminalNode` requires parser instance which is not available in PostProcessor.

Reply:
We should pass the parser in the constructor of the postprocessor to use here, then.